workflows/detaxizer.nf

```diff
@@ -1,129 +1,480 @@
+include { softwareVersionsToYAML } from '../subworkflows/nf-core/utils_nfcore_pipeline'
+include { methodsDescriptionText } from '../subworkflows/local/utils_nfcore_detaxizer_pipeline'
 
-// Print parameter summary log to screen
-log.info logo + paramsSummaryLog(workflow) + citation
-
-WorkflowDetaxizer.initialise(params, log)
+include { FASTP             } from '../modules/nf-core/fastp/main'
+include { KRAKEN2_KRAKEN2   } from '../modules/nf-core/kraken2/kraken2/main'
+include { BLAST_BLASTN      } from '../modules/nf-core/blast/blastn/main'
+include { BLAST_MAKEBLASTDB } from '../modules/nf-core/blast/makeblastdb/main'
 
+include { RENAME_FASTQ_HEADERS_PRE              } from '../modules/local/rename_fastq_headers_pre'
+include { KRAKEN2PREPARATION                    } from '../modules/local/kraken2preparation'
+include { PARSE_KRAKEN2REPORT                   } from '../modules/local/parse_kraken2report'
+include { ISOLATE_IDS_FROM_KRAKEN2_TO_BLASTN    } from '../modules/local/isolate_ids_from_kraken2_to_blastn'
+include { PREPARE_FASTA4BLASTN                  } from '../modules/local/prepare_fasta4blastn'
+include { FILTER_BLASTN_IDENTCOV                } from '../modules/local/filter_blastn_identcov'
+include { FILTER                                } from '../modules/local/filter'
+include { RENAME_FASTQ_HEADERS_AFTER            } from '../modules/local/rename_fastq_headers_after'
+include { SUMMARY_KRAKEN2                       } from '../modules/local/summary_kraken2'
+include { SUMMARY_BLASTN                        } from '../modules/local/summary_blastn'
+include { SUMMARIZER                            } from '../modules/local/summarizer'
 /*
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-    CONFIG FILES
+    RUN MAIN WORKFLOW
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 */
 
-ch_multiqc_config          = Channel.fromPath("$projectDir/assets/multiqc_config.yml", checkIfExists: true)
-ch_multiqc_custom_config   = params.multiqc_config ? Channel.fromPath( params.multiqc_config, checkIfExists: true ) : Channel.empty()
-ch_multiqc_logo            = params.multiqc_logo   ? Channel.fromPath( params.multiqc_logo, checkIfExists: true ) : Channel.empty()
-ch_multiqc_custom_methods_description = params.multiqc_methods_description ? file(params.multiqc_methods_description, checkIfExists: true) : file("$projectDir/assets/methods_description_template.yml", checkIfExists: true)
+// speficy the fasta parameter if it is not provided via --fasta
+def fasta = false
 
-/*
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-    IMPORT LOCAL MODULES/SUBWORKFLOWS
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-*/
+if (!params.fasta) {
+    // If params.fasta is false and params.genome is present
+    if (params.genome) {
+        fasta = params.genomes[params.genome]?.fasta ?: false
```

Comment:
I don't think I've seen that syntax used before. What does `params.genomes[params.genome]?.fasta` evaluate to?

Reply:
it evaluates to the path for the fasta used to build the blastn DB, from `igenomes.conf` (as I understand it at the moment IF the fasta entry is present in the `igenomes.conf`, but I cannot recall at the moment why I did it this way ... )

Reply:
Normally the getGenomeAttributesFunction should take care of this. 🤨